src/RWA007-A/scripts/RWA007Deployment.s.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.6.12;
+
+import "forge-std/Script.sol";
+import "../contract/Import.sol";
+
+interface Changelog {
+    function getAddress(bytes32 what) external returns (address);
+}
+
+interface JoinFab {
+    function newAuthGemJoin(address owner, bytes32 ilk, address gem) external returns (address);
+}
+
+contract RWA007Deployment is Script {
+    address immutable MCD_PAUSE_PROXY;
+    address immutable MCD_DAI;
+    address immutable MCD_VAT;
+    address immutable MCD_JUG;
+    address immutable MCD_JOIN_DAI;
+    address immutable MCD_PSM_USDC_A;
+    RwaTokenFactory immutable RWA_TOKEN_FAB;
+    JoinFab immutable JOIN_FAB;
+    Changelog CHANGELOG;
+
+    string NAME;
+    string SYMBOL;
+    string LETTER;
+    string SYMBOL_LETTER;
+
+    bytes32 immutable ILK;
+
+    event Result(address RwaToken);
+    constructor() public {
+        CHANGELOG                 = Changelog(getEnvAddressRequired("CHANGELOG"));
+        MCD_PAUSE_PROXY           = CHANGELOG.getAddress(bytes32("MCD_PAUSE_PROXY"));
+        MCD_DAI                   = CHANGELOG.getAddress(bytes32("MCD_DAI"));
+        MCD_VAT                   = CHANGELOG.getAddress(bytes32("MCD_VAT"));
+        MCD_JUG                   = CHANGELOG.getAddress(bytes32("MCD_JUG"));
+        MCD_JOIN_DAI              = CHANGELOG.getAddress(bytes32("MCD_JOIN_DAI"));
+        MCD_PSM_USDC_A            = CHANGELOG.getAddress(bytes32("MCD_PSM_USDC_A"));
+        RWA_TOKEN_FAB             = RwaTokenFactory(CHANGELOG.getAddress(bytes32("RWA_TOKEN_FAB")));
+        JOIN_FAB                  = JoinFab(CHANGELOG.getAddress(bytes32("JOIN_FAB")));
+
+        NAME                      = getEnvStringRequired("NAME");
+        SYMBOL                    = getEnvStringRequired("SYMBOL");
+        LETTER                    = getEnvStringRequired("LETTER");
+        SYMBOL_LETTER             = string(abi.encodePacked(SYMBOL, "_", LETTER));
+
+        ILK                       = bytesToBytes32(abi.encodePacked(SYMBOL, string("-"), LETTER));
+
+    }
+
+    function run() external {
```

Comment:
Did you try to execute this script?
I wonder what would happen if we ever hit the block size limit.

We'd probably need a way to split `run()` into several steps. Not sure how to do that though. I don't think script contracts can have state.

Reply:
I think the way it's works, that forge collect all transactions and then sent them separately, so we won't hit block limit

Reply:
Actually, maybe you are right, forge simulates this method internally, and there it can happen. Maybe we can leave this for now and adjust `deploy.sh` to store state in ENV vars when we get to that problem in the future?